castra/core.py

```diff
@@ -196,9 +196,12 @@ def load_partition(self, name, columns, categorize=True):
         index = unpack_file(self.dirname(name, '.index'))
 
         df = pd.DataFrame(dict(zip(columns, arrays)),
-                          columns=pd.Index(columns, name=self.axis_names[1]),
-                          index=pd.Index(index, dtype=self.index_dtype,
-                                         name=self.axis_names[0]))
+                          columns=pd.Index(columns, name=self.axis_names[1],
+                                           tupleize_cols=False),
```

Comment:
`tupleize_cols` is a little strange, when it's `True` pandas calls `sorted` on the input if it's a `list`, which is inefficient when you have known unordered categories. Right now, we don't handle unordered categories in castra, so this saves some non trivial number of cycles